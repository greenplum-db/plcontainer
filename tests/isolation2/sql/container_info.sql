CREATE OR REPLACE FUNCTION hello() RETURNS text AS $$
# container: plc_python_shared
return "hello"
$$ LANGUAGE plcontainer;

1: SELECT hello();

2: SELECT "UP_TIME" FROM plcontainer_containers_info();

SELECT gp_inject_fault_infinite('before_get_container_info', 'reset', dbid)
  FROM gp_segment_configuration WHERE role='p' AND content= -1;

SELECT gp_inject_fault_infinite('before_get_container_info', 'suspend', dbid)
  FROM gp_segment_configuration WHERE role='p' AND content= -1;

2&: SELECT "UP_TIME", "MEMORY_USAGE(KB)", "CPU_USAGE", "UDF INFO" FROM plcontainer_containers_info();

1: SELECT pg_terminate_backend(pg_backend_pid());

SELECT gp_inject_fault_infinite('before_get_container_info', 'resume', dbid)
  FROM gp_segment_configuration WHERE role='p' AND content= -1;

2<:
