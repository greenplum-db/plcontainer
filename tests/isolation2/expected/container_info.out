CREATE OR REPLACE FUNCTION hello() RETURNS text AS $$ # container: plc_python_shared return "hello" $$ LANGUAGE plcontainer;
CREATE FUNCTION

1: SELECT hello();
 hello 
-------
 hello 
(1 row)

2: SELECT "UP_TIME" FROM plcontainer_containers_info();
 UP_TIME               
-----------------------
 Up Less than a second 
(1 row)

SELECT gp_inject_fault_infinite('before_get_container_info', 'reset', dbid) FROM gp_segment_configuration WHERE role='p' AND content= -1;
 gp_inject_fault_infinite 
--------------------------
 Success:                 
(1 row)

SELECT gp_inject_fault_infinite('before_get_container_info', 'suspend', dbid) FROM gp_segment_configuration WHERE role='p' AND content= -1;
 gp_inject_fault_infinite 
--------------------------
 Success:                 
(1 row)

2&: SELECT "UP_TIME", "MEMORY_USAGE(KB)", "CPU_USAGE", "UDF INFO" FROM plcontainer_containers_info();  <waiting ...>

1: SELECT pg_terminate_backend(pg_backend_pid());
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.

SELECT gp_inject_fault_infinite('before_get_container_info', 'resume', dbid) FROM gp_segment_configuration WHERE role='p' AND content= -1;
 gp_inject_fault_infinite 
--------------------------
 Success:                 
(1 row)

2<:  <... completed>
 UP_TIME | MEMORY_USAGE(KB) | CPU_USAGE | UDF INFO 
---------+------------------+-----------+----------
 Removed |                  |           |          
(1 row)
